src/main/webapp/room.js

```diff
@@ -0,0 +1,19 @@
+let players = [];
+
+const createUser = (ev)=>{
+    ev.preventDefault();  //to stop the form submitting
+    let user = {
+        username: document.getElementById('new-player-id').value,
+        get userID() {
```

Comment:
I think this means the ID will change each time the userID() method is called. This may be a problem if you e.g. store a user ID in a cookie and then try to look up the user by ID from the players array. The second time you call userID() will return a different ID than the first time you called it.

Instead, I think you can do something like you did with username:
userID : '_' + Math.random().toString(36).substr(2, 9);

That way the user ID is set once and doesn't change afterwards. However I'm not too familiar with JavaScript so if that doesn't work, let me know and we can figure out what to do.

Reply:
+1